kopykat-gradle-plugin/src/main/kotlin/fp/serrano/kopykat/KopyKatPlugin.kt

```diff
@@ -0,0 +1,97 @@
+package fp.serrano.kopykat
+
+import com.google.devtools.ksp.gradle.KspExtension
+import com.google.devtools.ksp.gradle.KspGradleSubplugin
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.create
+import org.gradle.kotlin.dsl.withType
+import org.gradle.plugins.ide.idea.IdeaPlugin
+import org.gradle.plugins.ide.idea.model.IdeaModel
+import org.jetbrains.kotlin.gradle.dsl.kotlinExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinBasePlugin
+
+abstract class KopyKatPlugin : Plugin<Project> {
+
+    override fun apply(target: Project) {
+
+        val kkSettings = createKopyKatSettings(target)
+
+        configureKsp(target, kkSettings)
+
+        target.afterEvaluate {
+            // normally afterEvaluate should be avoided, but it's necessary here to allow KopyKatSettings to be
+            // configured otherwise applyKspPlugin will always be the default value.
+            if (kkSettings.applyKspPlugin.getOrElse(false)) {
+                target.pluginManager.apply("com.google.devtools.ksp")
+            }
+        }
+    }
+
+    private fun createKopyKatSettings(target: Project): KopyKatSettings {
+        return target.extensions.create<KopyKatSettings>(KOPYKAT_EXTENSION_NAME).apply {
+            applyKspPlugin.convention(true)
+
+            mutableCopy.convention(true)
+            copyMap.convention(true)
+            hierarchyCopy.convention(true)
+        }
+    }
+
+    private fun configureKsp(target: Project, kkSettings: KopyKatSettings) {
+        target.plugins.withType<KspGradleSubplugin> {
+
+            target.configureSourceSets()
+
+            target.configurations.matching { it.name == KspGradleSubplugin.KSP_MAIN_CONFIGURATION_NAME }.configureEach {
+                defaultDependencies {
+                    add(target.dependencies.create("com.github.kopykat-kt.kopykat:ksp:1.0-rc1"))
+                }
+            }
```

Comment:
defining a dependency like this doesn't seem to work. It might need to be a mandatory dependency. I'll investigate.